bin/spectacle/init.js

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env node
+
+'use strict';
+
+const prompts = require('prompts');
+
+const mdRegex = /(\.mdx$|\.md$)/gm;
+const jsRegex = /(\.json$|\.js$)/gm;
+const jsxRegex = /(\.jsx$|\.js$)/gm;
+const { pathExists } = require('fs-extra');
+
+const questions = [
+  {
+    type: 'select',
+    name: 'action',
+    message: 'Create a live server or build directory?',
+    choices: [
+      { title: 'Server', value: 'server' },
+      { title: 'Build', value: 'build' }
+    ]
+  },
+  {
+    type: prev => (prev === 'server' ? 'number' : null),
+    name: 'port',
+    message: 'What port for the live server?',
+    validate: val => {
+      const numLength = val.toString().length;
+      // eslint-disable-next-line no-magic-numbers
+      if (numLength === 4) {
+        return val;
+      }
+      return 'The port should be 4 numbers';
+    }
+  },
+  {
+    type: 'text',
+    name: 'src',
+    message:
+      'What is the file directory of the markdown source? Include the file extension',
+    validate: async filepath => {
+      const filePathExists = await pathExists(filepath);
+      if (!filePathExists || !filepath.match(mdRegex)) {
+        return 'Error, cannot find that file at the path specified. The file must have a .md or .mdx extension.';
+      }
+      return filePathExists && filepath.match(mdRegex);
+    }
+  },
+  {
+    type: 'confirm',
+    name: 'custom_theme',
+    message: 'Do you have a custom theme file?',
+    initial: false
+  },
+  {
+    // eslint-disable-next-line no-constant-condition
+    type: prev => (prev ? 'text' : null),
+    name: 'theme',
+    message:
+      'What is the file directory of the custom theme file? Include the file extension',
+    validate: async filepath => {
+      const filePathExists = await pathExists(filepath);
+      if (!filePathExists || !filepath.match(jsRegex)) {
```

Comment:
Can theme actually handle being a JSX file? Just looking at template which does...

Reply:
This one is a .js or .json file, not a .jsx file.

Reply:
Reading through the webpack code, we actually would pass this to babel I think so it could be JSX, but it isn't documented as such, and there's not a lot of reason in current form to open it up to JSX. But, we can in the future!

... well, and I guess not a lot of teams even use the `.jsx` extension much anymore 😂